src/test/java/com/zufar/icedlatte/favorite/api/endpoint/FavoriteListEndpointTest.java

```diff
@@ -0,0 +1,170 @@
+import static com.zufar.icedlatte.test.config.RestAssertion.assertRestApiEmptyBodyResponse;
+import static com.zufar.icedlatte.test.config.RestAssertion.assertRestApiOkResponse;
+import static io.restassured.RestAssured.given;
+
+@Testcontainers
+@DisplayName("FavoriteListEndpointTest Tests")
+@ActiveProfiles("test")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class FavoriteListEndpointTest {
+
+    @Container
+    @ServiceConnection
+    static PostgreSQLContainer<?> postgres = new PostgreSQLContainer<>("postgres:13.11-bullseye");
+
+    @LocalServerPort
+    protected Integer port;
+
+    @Value("${jwt.secret}")
+    protected String secretKey;
+
+    @Value("${jwt.expiration}")
+    protected Long expiration;
+
+    @Value("${jwt.email}")
+    protected String email;
+
+    @Value("${jwt.password}")
+    protected String password;
+
+    private static final String AUTHENTICATE_TEMPLATE = "/security/model/authenticate-template.json";
+
+    private static final String FAVORITE_LIST_SCHEMA = "favorite/model/schema/favorite-list-schema.json";
+    private static final String FAVORITE_LIST_ERROR_SCHEMA = "favorite/model/schema/favorite-list-error-schema.json";
+    private static final String PRODUCT_ADD_TO_FAVORITE_LIST = "/favorite/model/product-add-to-favorite-list.json";
+    private static final String PRODUCT_NOT_EXIST_ADD_TO_FAVORITE_LIST = "/favorite/model/product-not-exist-add-to-favorite-list.json";
+
+    protected static RequestSpecification specification;
+
+    @BeforeEach
+    void tokenAndSpecification() {
+        String jwtToken = getJwtToken();
+        specification = given()
+                .log().all(true)
+                .port(port)
+                .header("Authorization", "Bearer " + jwtToken)
+                .basePath(FavoritesEndpoint.FAVORITES_URL)
+                .contentType(ContentType.JSON)
+                .accept(ContentType.JSON);
+    }
+
+    protected String getRequestBody(String resourcePath) {
+        try {
+            JsonNode json = JsonLoader.fromResource(resourcePath);
+            return json.toString();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected String getJwtToken() {
+        specification = given()
+                .log().all(true)
+                .port(port)
+                .basePath(UserSecurityEndpoint.USER_SECURITY_API_URL)
+                .contentType(ContentType.JSON)
+                .accept(ContentType.JSON);
+
+        String body = getRequestBody(AUTHENTICATE_TEMPLATE)
+                .formatted(email, password);
+
+        Response response = given(specification)
+                .body(body)
+                .post("/authenticate");
+
+        String jwtToken = response.getBody().path("token");
+
+        if (isJwtTokenNotValid(jwtToken)) {
+            throw new RuntimeException("JWT Token is empty or null. Test failed.");
```

Comment:
be more consice and choose not a generic exception. At least you can throw IllegalArgumentException instead